feat(core): add loadable solana tokens (#92)
vacuumlabs/trezor-firmware: docs/common/solana-definitions.md

# Solana definitions

To allow nicer token presentation on the device, Trezor allows passing Definitions as additional info. Currently only
one token can be included in the definitions. Solana tokens are uniquely identified by their program id and mint

> account, Trezor additionally expects a name and a ticker. Additionally in the future we'll be passing additional

Reviewer: nitpick

```suggestion
account, Trezor additionally expects a name and a ticker.

In the future we might also include definitions for instructions.
```

From this list, a collection of binary blobs is generated, signed, and made available online.

A given Trezor firmware will only accept signed definitions newer than a certain date, typically one month before
firmware release. This means that a client application should either always fetch fresh definitions from the official
URLs, or refresh its local copy frequently.

The base URL for the definitions is `https://data.trezor.io/firmware/solana-definitions/`.

## Definition format

Each definition is encoded as a protobuf message `SolanaTokenInfo` and packaged in the following binary format.

All numbers are unsigned little endian.

1. magic string `trzd1` (5 bytes)
2. definition type according to `DefinitionType` enum (1 byte)
3. data version of the definition (4 bytes)
4. protobuf payload length (2 bytes)
5. protobuf payload (N bytes)

A Merkle tree is constructed from all binary definitions (see below) and its root is signed by the CoSi algorithm.

The full format of the definition is as follows:

1. Data payload (see above)
2. Number of Merkle proof entries (1 byte)
3. Sequence of 32-byte proof entries (N * 32 bytes)
4. CoSi sigmask (1 byte)

> 5. CoSi signature (64 bytes)

Reviewer: Maybe we can just link to `./ethereum-definitions.md`? Or perhaps introducing a `definitions.md` with common information might also make sense, but I find the link a bit better. (same for "Merkle tree algorithm" section)
